src/pivot-table/hooks/use-column-width.ts

```diff
@@ -96,57 +99,79 @@ export default function useColumnWidth(
+
+      const lastRow = headersData.data[headersData.size.y - 1] as HeaderCell[];
+      const columnWidths = lastRow.map((cell, index) => {
+        let width: number;
+
+        if (widthOverride && overrideIndex !== undefined && overrideIndex === index) {
+          width = widthOverride;
+        } else if (cell.id === PSEUDO_DIMENSION_KEY) {
+          // Use the max width of all measures
+          width = Math.max(
+            ...qMeasureInfo.map(({ qFallbackTitle, columnWidth }) => {
+              const fitToContentWidth = measureTextForDimensionValue(qFallbackTitle) + TOTAL_CELL_PADDING;
+              return getColumnWidth(columnWidth, fitToContentWidth);
+            }),
+          );
+        } else {
+          const { label, qApprMaxGlyphCount, columnWidth, isLocked } = cell;
+          const expandIconSize = !isFullyExpanded && index < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
+          const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
+
+          const fitToContentWidth = Math.max(
+            measureTextForHeader(label) + TOTAL_CELL_PADDING + MENU_ICON_SIZE + lockedIconSize,
+            estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + expandIconSize,
+          );
 
-      if (cell.id === PSEUDO_DIMENSION_KEY) {
-        // Use the max width of all measures
-        width = Math.max(
-          ...qMeasureInfo.map(({ qFallbackTitle, columnWidth }) => {
-            const fitToContentWidth = measureTextForDimensionValue(qFallbackTitle) + TOTAL_CELL_PADDING;
-            return getColumnWidth(columnWidth, fitToContentWidth);
-          }),
-        );
-      } else {
-        const { label, qApprMaxGlyphCount, columnWidth, isLocked } = cell;
-        const expandIconSize = !isFullyExpanded && index < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
-        const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
+          width = getColumnWidth(columnWidth, fitToContentWidth);
+        }
 
-        const fitToContentWidth = Math.max(
-          measureTextForHeader(label) + TOTAL_CELL_PADDING + MENU_ICON_SIZE + lockedIconSize,
-          estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + expandIconSize,
-        );
+        sumOfWidths += width;
+        return width;
+      });
 
-        width = getColumnWidth(columnWidth, fitToContentWidth);
-      }
+      return {
+        leftGridWidth: Math.min(rect.width * LEFT_GRID_MAX_WIDTH_RATIO, sumOfWidths),
+        leftGridColumnWidths: columnWidths,
+        leftGridFullWidth: sumOfWidths,
+      };
+    },
+    [
+      estimateWidthForDimensionValue,
+      headersData.data,
+      headersData.size.y,
+      isFullyExpanded,
+      measureTextForDimensionValue,
+      measureTextForHeader,
+      qMeasureInfo,
+      qNoOfLeftDims,
+      rect.width,
+    ],
+  );
 
-      sumOfWidths += width;
-      return width;
-    });
+  const [leftGridWidthInfo, setLeftGridWidthInfo] = useState<LeftGridWidthInfo>(calculateLeftGridWidthInfo);
```

Comment:
This initial state is the function? Should this not be `useState<LeftGridWidthInfo>(calculateLeftGridWidthInfo());`?

Reply:
no, that is not the syntax, quite confusing. If you pass a function to setState, it runs the function and sets the state to the return value. In order to set the state to function, you have to do `setState(() => nameOfFunction)`

Reply:
maybe to prevent the confusion around this, it would be better to keep the () after the function name, if you look into this line and then look into line 174 in useEffect, definitely you will get confused

Reply:
I'd say it is pretty basic hook knowledge ;) But I'll add it, no need for unnecessary confution

Reply:
It's actually 2 different uses. By doing `useState<LeftGridWidthInfo>(calculateLeftGridWidthInfo);` the function is only called the first the time the hook is "rendered". With `useState<LeftGridWidthInfo>(calculateLeftGridWidthInfo());` it's called on every render, which is probably not what we want, nor is it going to be better from a performance spective.

IMO change it back and add a comment instead.

I will try and do some quick testing on the impact of it.

Reply:
Some quick testing: `calculateLeftGridWidthInfo` is called 4 times as much now compared to before when re-sizing a column

Reply:
oh, I didn't see this discussion. I'll change it back :)